docs/rfc-identity.md

```diff
@@ -0,0 +1,186 @@
+
+Because the identity was linked to this device at setup time, and depending on the DID method, profile information and verifiable claims might become stale. For this reason, users will be able to sync up with their identity to update any stale data.
+
+### Managing Verifiable Claims
+
+Users will be able to add claims to their identities. Initially, we will focus on adding self-signed claims linked to some of the most popular social networks. The process of adding those claims (and proofs) will work similarly to [Keybase](https://keybase.io/) in the sense that users will be guided throughout the process.
+
+The exact way in how the claims will be stored is dependent on the DID method. For instance, uPort already provides a way to store these claims via attestations.
+
+### Authentication on applications
+
+Peer-Star applications will use the IdentityManagerClient to manage the user's session.
+
+When bootstrapped, the IdentityManagerClient will attempt to read the Session Public Key from the local storage of the applications' origin. The IdentityManagerClient then takes the Session Public Key and queries the IdentityManager to retrieve the session data associated to it. If the IdentityManager responds back with the session data, the user is authenticated and, as such, there's an Authenticated Session. If the IdentityManager doesn't recognize that Session Public Key or if there's no Session Public Key in the first place, there's no Authenticated Session, meaning that no user is authenticated. In such cases, the application may request the Identity Manager to authenticate the user, typically via a login button.
+
+When the application requests the IdentityManagerClient to authenticate the user, a popup pointing to the IdentityManager authenticate screen is open (via `window.open`). In this screen, the user is asked to choose an identity and to disclose some information, such as its name, its photo and a set of claims & proofs. If the user consented to the disclosure of this information and entered the Device Private Key passphrase correctly, a new session for this application will be created and stored. The session data, including the Session Public Key and its signature signed by the Device Private Key, is sent back to the application. If the user denied the disclosure of the information, the operation will fail.
+In both scenarios, the popup is closed and the outcome is made available to the application.
+
+Applications might choose the TTL of a session depending on the degree of security they want to have.
+
+### Signing artifacts on applications
+
+Applications may want to sign artifacts, such as regular data or Ephemeral Keys. This will be possible in two different ways:
+
+1. Sign with the Session Private Key
+2. Sign with the Device Private Key
```

Comment:
Could a third option be to issue a one-time-use key? I've used [json web tokens](https://jwt.io) in the past for such a purpose back when they were considered a good idea.

For this to work one would need to scope the capabilities of such a key, or place a reasonably-short TTL on the key.

Reply:
Will answer below as it is related.